add from_sparse method

trsigg opened the pull request:
Method for creating a `SparseBlockArray` from a `scipy.sparse` object.

trsigg committed:
add from_sparse method

dlzou commented:
@trsigg Can you remove the `asscalar` fallback so tests pass? Should be the same as #563. Also, can you add a test case for the new method?

trsigg committed:
remove asscalar fallback

trsigg committed:
add new test and outward API, fix old test

- add COO creation support in `ArrayApplication.array` and outward-facing `nps.from_coo` method
- add `test_from_coo`
- fix bug in `test_sparse_init` where expected results were for `int64` but arrays were created as `int32`
